### PR TITLE
protocol type added

### PR DESCRIPTION
**Pull request by jonathanMweiss**
Adding a protocol field to the trackingID

### CODE REVIEW

File: io.pb.go [hunk 1]
```diff
@@ -187,6 +187,7 @@ type TrackingID struct {
 	PartiesState []byte `protobuf:"bytes,2,opt,name=parties_state,json=partiesState,proto3" json:"parties_state,omitempty"`
 	// any auxilary data provided to the protocol from outside, and needs to be on every message.
 	AuxilaryData  []byte `protobuf:"bytes,3,opt,name=auxilary_data,json=auxilaryData,proto3" json:"auxilary_data,omitempty"`
+	Protocol      []byte `protobuf:"bytes,4,opt,name=protocol,proto3" json:"protocol,omitempty"` // defines the protocol type.
```
Comment (yossigi) [inline io.pb.go]: shouldn't the protocol field be just one byte?

File: util_test.go [hunk 2]
```diff
@@ -118,3 +124,128 @@ func TestTrackingIdFromString(t *testing.T) {
...
+	s := "01-" + hx(ProtocolEmpty.ToString()) + "--"
+	err := tid.FromString(s)
+	if !errors.Is(err, errNilTrackID) {
+		t.Fatalf("FromString(nil receiver) err=%v, want errNilTrackID", err)
+	}
+}
+
+// // --- Negative/edge cases for FromString ------------------------------------
+
+// func TestFromString_Errors(t *testing.T) {
```
Comment (yossigi) [inline util_test.go]: no need for this dead code test? Should it be deleted or revived?

File: util.go [hunk 3]
```diff
@@ -94,19 +100,24 @@ func (t *TrackingID) FromString(s string) error {
...
-	if len(parts) != 3 {
+	if len(parts) != 4 {
 		return errTrackidInvalidFormat
 	}
 
 	if len(parts[0]) == 0 {
 		return errTrackidMustHaveDigest
 	}
 
+	if len(parts[1]) == 0 {
```
Comment (yossigi) [inline util.go]: should it test that this part has length of exactly 1? And for the digest above, that the tracking number has the right digest length?

**Review by yossigi — APPROVED** [review]
